_search-plugins/searching-data/search-shard-routing.md

```diff
@@ -0,0 +1,212 @@
+
+    ```json
+    GET /my-index/_search?preference=_prefer_nodes:node-1,node-2
+    ```
+    {% include copy-curl.html %}
+
+7. `_shards:<shard-id>,<shard-id>`: Limits the search to specific shards.
+
+    ```json
+    GET /my-index/_search?preference=_shards:0,1
+    ```
+    {% include copy-curl.html %}
+
+8. `_local`: Executes the search on the local node if the data is available there, which can reduce latency.
+
+    ```json
+    GET /my-index/_search?preference=_local
+    ```
+    {% include copy-curl.html %}
+
+9. Custom string: You can use any custom string as the preference value. This custom string ensures that requests containing the same string are routed to the same shards consistently, which can be useful for caching.
+
+    ```json
+    GET /my-index/_search?preference=custom_string
+    ```
+    {% include copy-curl.html %}
+
+## Custom routing during index and search
+
+You can specify routing during both indexing and search operations.
+
+### Routing during indexing
+When you index a document, OpenSearch calculates a hash of the routing value and uses this hash to determine the shard on which the document will be stored. If you don't specify a routing value, OpenSearch uses the document's ID to calculate the hash.
+
+The following is an example index operation with a routing value:
+
+```json
+POST /index1/_doc/1?routing=user1
+{
+  "name": "John Doe",
+  "age": 20
+}
+```
+{% include copy-curl.html %}
+
+In this example, the document with ID 1 is indexed with the routing value user1. All documents with the same routing value will be stored on the same shard.
+
+### Routing during searches
+
+When you search for documents, specifying the same routing value ensures that the search request is routed to the appropriate shard. This can significantly improve performance by reducing the number of shards that need to be queried.
+
+Here’s an example of searching with a specific routing value:
+
+```json
+GET /index1/_search?routing=user1
+{
+  "query": {
+    "match": {
+      "name": "John Doe"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In this example, the search query is routed to the shard that contains documents indexed with the routing value `user1`.
+
+Additional consideration needs to be taken when using custom routing in order to prevent hot spots and data skew:
+
+ - A _hot spot_ occurs when a disproportionate number of documents are routed to a single shard. This can lead to that shard becoming a bottleneck, because it will have to handle more read and write operations compared to other shards. Consequently, this shard may experience higher CPU, memory, and I/O usage, leading to performance degradation.
+
+ - _Data skew_ refers to an uneven distribution of data across shards. If routing values are not evenly distributed, some shards may end up storing significantly more data than others. This can result in an imbalanced storage use, where certain nodes have a much higher disk utilization compared to others.
```

Comment:
```suggestion
 - _Data skew_ refers to an uneven distribution of data across shards. If routing values are not evenly distributed, some shards may end up storing significantly more data than others. This can result in imbalanced storage usage, where certain nodes have a much higher disk utilization compared to others.
```